src/main/scala/com/fulcrumgenomics/util/ReadStructure.scala

```diff
@@ -24,88 +24,136 @@
-  case class SubRead(bases: String, quals: Option[String] = None, segment: ReadSegment)
+    segs
+  }
 
   /**
-    * Inserts square brackets around the characers in the read structure that are causing the error.
+    * Inserts square brackets around the characters in the read structure that are causing the error.
     *
-    * @param s the read structure string
+    * @param rs the read structure string
     * @param start the start of the error in the string (inclusive)
     * @param end the ned of the error in the string (exclusive)
     */
-  private def formatReadStructureError(s: String, start: Int, end: Int) = {
-    val prefix = s.substring(0, start)
-    val error  = s.substring(start, end)
-    val suffix = if (end == s.length) "" else s.substring(end, s.length)
-    s"$prefix[$error]$suffix"
+  private def invalid(msg: String, rs: String, start: Int, end: Int): Nothing = {
+    val prefix = rs.substring(0, start)
+    val error  = rs.substring(start, end)
+    val suffix = if (end == rs.length) "" else rs.substring(end, rs.length)
+    throw new IllegalArgumentException(s"$msg: $prefix[$error]$suffix")
+  }
+
+  /**
+    * Creates sub-reads for the set of read segments provided.
+    * @param bases the sequenced bases
+    * @param segments the set of segments to extract
+    * @return a sequence of [[SubRead]] objects, one for each segment
+    */
+  def extract(bases: String, segments: Seq[ReadSegment]): Seq[SubReadWithoutQuals] = segments.map(_.extract(bases))
+
+  /**
+    * Creates sub-reads for the set of read segments provided.
+    * @param bases the sequenced bases
+    * @param quals the quality scores for the bases (must be same length as bases)
+    * @param segments the set of segments to extract
+    * @return a sequence of [[SubRead]] objects, one for each segment
+    */
+  def extract(bases: String, quals: String, segments: Seq[ReadSegment]): Seq[SubReadWithQuals] = {
+    assert(bases.length == quals.length)
+    segments.map(s => s.extract(bases, quals))
   }
 
   /** Splits the given bases into tuples with its associated read segment.  If strict is false then only return
-    * tuples for which we have bases in `bases`, otherwise throw an exception.
+    * [[com.fulcrumgenomics.util.ReadStructure.SubRead]]s for which we have bases in `bases`, otherwise throw an exception.
     **/
-  def structureRead(bases: String, segments: Seq[ReadSegment], strict: Boolean = true): Seq[SubRead] = {
-    // TODO: TF: strict == false allows us to skip segments (in the filter)
-    segments.filter(s => strict || s.offset < bases.length).map {
-      segment =>
-        val (start, end) = segment.range(bases.length, strict=strict)
-        SubRead(bases=bases.substring(start, end), segment=ReadSegment(segment, end - start))
-    }
+  @deprecated(message="Use ReadStructure.extract() instead.")
+  def structureRead(bases: String, segments: Seq[ReadSegment]): Seq[SubReadWithoutQuals] = {
```

Comment:
Since both the parameters and return type have changed, I am in favor of removing the previous method altogether!